spec/options_spec.rb

```diff
@@ -1,40 +1,150 @@
 # typed: false
 describe RedisMemo::Options do
   context 'cache validation' do
-    it 'validates the cache result' do
-      allow(RedisMemo::DefaultOptions).to receive(:cache_validation_sampler) do
-        proc { true }
-      end
+    def allow_rand_return(value)
+      allow(Random).to receive(:rand).and_return(value)
+    end
 
+    def expect_count(orig_value, updated_value)
       count = 0
 
       allow(RedisMemo::DefaultOptions).to receive(:cache_out_of_date_handler) do
         proc { count += 1 }
       end
 
-      klass = Class.new do
-        extend RedisMemo::MemoizeMethod
-        attr_accessor :count
-
-        def calc
-          @count += 1
-        end
-
-        memoize_method :calc
-      end
-
       obj = klass.new
       obj.count = 0
 
       # cache miss
       expect {
         obj.calc
-      }.to change { count }.by(0)
+      }.to change { count }.by(orig_value)
 
       # cache hit
       expect {
         obj.calc
-      }.to change { count }.by(1)
+      }.to change { count }.by(updated_value)
+    end
+
+    def expect_cache_validation
+      # With cache validation, the count will change by 1
+      expect_count(0, 1)
+    end
+
+    def expect_no_cache_validation
+      # Without cache validation, the count will not change
+      expect_count(0, 0)
+    end
+
+    context 'with only global cache validation sample percentage' do
+      def set_cache_validation_sample_percentage(value)
+        allow(RedisMemo::DefaultOptions).to receive(:cache_validation_sample_percentage).and_return(value)
+      end
+
+      let!(:klass) do
+        klass = Class.new do
```

Comment:
```suggestion
        Class.new do
```
this is causing a build error